### PR TITLE
UAR-1479 Add CSRF Mitigation for OE registration journey

### PR DESCRIPTION
**Pull request by dlloyd1-ch**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-1479

### Change description
- CSRF token implementation to prevent CSRF attacks. Includes:
- Implementation of CSRF middleware. Note this is disabled in order for the tests to pass.
- Implementation of CSRF token in the OE Registration journey.
- Updated the corresponding OE registration tests to mock out the CSRF middleware.

#### Steps to implement for the whole of overseas-entities-web
- Enable csrfProtectionMiddleware 
- Add the CSRF token to each page with a `<form method="post">` element.
- Update the corresponding unit tests to mock out the CSRF middleware

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -62,6 +64,13 @@ const cookieConfig = {
 const sessionStore = new SessionStore(new Redis(`redis://${config.CACHE_SERVER}`));
 app.use(SessionMiddleware(cookieConfig, sessionStore));
 
+const csrfProtectionMiddleware = CsrfProtectionMiddleware({
+  sessionStore,
+  enabled: false,
```
Comment (mwestacott) [inline src/app.ts:69]: Probably worth adding a TODO comment in the code at this point to make clear that this middleware will be enabled when all journeys implemented (which I think is the case).

File: src/app.ts [hunk 2]
```diff
@@ -29,6 +30,7 @@ const nunjucksEnv = nunjucks.configure([
   "views/update/remove",
   "node_modules/govuk-frontend/",
   "node_modules/govuk-frontend/components",
+  "node_modules/@companieshouse/"
```
Comment (mwestacott) [inline src/app.ts:33]: Just wondering if / on the end is required? Seems to be there in some lines above and not others. Maybe we can save a byte!

File: views/update/update-layout.html [hunk 3]
```diff
@@ -20,6 +20,7 @@
 {% from "govuk/components/select/macro.njk"              import govukSelect %}
 {% from "govuk/components/warning-text/macro.njk"        import govukWarningText %}
 {% from "govuk/components/notification-banner/macro.njk" import govukNotificationBanner %}
+{% from "web-security-node/components/csrf-token-input/macro.njk" import csrfTokenInput %}
```
Comment (mwestacott) [inline views/update/update-layout.html:23]: Looks like this PR is for the registration journey but this change relates to Update. If possible, I'd revert this and make the change when we do the PR for the Update journey.

Comment (makhtar-ch) [reply views/update/update-layout.html:23]: @mwestacott Without this, 2 GET tests `"render the beneficial owner statements page with Update data"`on line 106 and 156 fail on beneficial.owner.statements.spec.ts

Comment (mwestacott) [reply views/update/update-layout.html:23]: Yep, sounds fine in that case.

**Review by markpit — APPROVED** [review]

**Review by dlloyd1-ch — COMMENTED** [review]
Happy with the PR, but I can't approve as I initially raised the PR for the PoC

**Review by mwestacott — APPROVED** [review]

**Pull request merged**